app/src/main/java/com/nextcloud/talk/upload/normal/FileUploader.kt

```diff
@@ -55,8 +55,10 @@ class FileUploader(
         try {
             val input: InputStream = context.contentResolver.openInputStream(sourceFileUri)!!
             val buf = ByteArray(input.available())
-            while (input.read(buf) != -1)
+            while (input.read(buf) != -1) {
                 requestBody = RequestBody.create("application/octet-stream".toMediaTypeOrNull(), buf)
+            }
+            input.close()
```

Comment:
```kotlin
            input.use {
                val buf = ByteArray(input.available())
                while (it.read(buf) != -1) {
                    requestBody = RequestBody.create("application/octet-stream".toMediaTypeOrNull(), buf)
                }
            }
```
`use` will auto-close the resource when the inner block is complete even if exceptions occur

Reply:
so just mark as false positive?

Reply:
ah, never mind, you posted the code piece 👍